app/views.py

```diff
@@ -8,3 +8,29 @@
 @App.route('/')
 def public_timeline():
     return render_template('public_timeline.html', flashes=db_utils.get_all_twittes())
+
+@app.route('/register', methods=['GET', 'POST']) # akef
+def register():
+    """Registers the user."""
+    if g.user:
+        return redirect(url_for('timeline'))
+    error = None
+    if request.method == 'POST':
+        if not request.form['username']:
+            error = 'You have to enter a username'
+        elif not request.form['email'] or \
+                '@' not in request.form['email']:
+            error = 'You have to enter a valid email address'
+        elif not request.form['password']:
+            error = 'You have to enter a password'
+        elif request.form['password'] != request.form['password2']:
+            error = 'The two passwords do not match'
+        elif get_user_id(request.form['username']) is not None:
+            error = 'The username is already taken'
+        else:
+            db = get_db()
+            register_user(request.form['username'],request.form['email'],request.form['password'])
+            db.commit()
```

Comment:
you don't need this also
as the register_user function executes this logic using db_execute
PLEASE READ THE CODE BEFORE USING IT